third_party/amd/lib/Dialect/TritonAMDGPU/IR/Dialect.cpp

```diff
@@ -46,3 +47,49 @@ void mlir::triton::amdgpu::TritonAMDGPUDialect::initialize() {
 
 #define GET_OP_CLASSES
 #include "Dialect/TritonAMDGPU/IR/Ops.cpp.inc"
+
+namespace mlir::triton::amdgpu {
+
+LogicalResult ViewSliceOp::verify() {
+  auto srcTy = getSource().getType();
+  auto srcLayout = srcTy.getEncoding();
+  auto srcElementType = dyn_cast<RankedTensorType>(srcTy).getElementType();
```

Comment:
You can use `mlir::getElementTypeOrSelf` here to handle scalar cases. Similarly for `resultElementType`. Though--I think we should error out for scalar cases?

Reply:
The source input is supposed to be an AnyRankedTensor. I was thinking that would take care of scalar cases. Is that not the case?